src/engraving/dom/arpeggio.cpp

```diff
@@ -81,8 +200,25 @@ void Arpeggio::editDrag(EditData& ed)
     double d = ed.delta.y();
     if (ed.curGrip == Grip::START) {
         m_userLen1 -= d;
+        PointF pos = PointF(chord()->canvasPos().x(), ed.pos.y());
+        EngravingItem* e = ed.view()->elementNear(pos);
+        if (e && e->isNote()) {
+            Chord* c = toNote(e)->chord();
+            int newSpan = m_span + track() - c->track();
+            undoChangeProperty(Pid::ARPEGGIO_SPAN, newSpan);
+            score()->undo(new ChangeParent(this, c, c->staffIdx()));
+            m_userLen1 = 0;
```

Comment:
Here too I wonder whether it is okay that `m_userLen1` is modified in a non-undoable way, but that was already the case before this PR...

Reply:
Yes - with the addition of `Arpeggio::startEditDrag` this works as expected!